### DISCUSSION TITLE
How to host .wasm file for client-side application in V2?

### DISCUSSION THREAD

**Discussion by bnopne** [hpcc-systems/hpcc-js-wasm#138] [General]
Hello!

I'm using `@hpcc-js/wasm` to generate DOT layouts in my **React** application. As for V1 i used `wasmFolder()` with custom URL to load wasm file, but there is not such function in V2, i have to `await Graphviz.load()` withour specifying URL.

What is preferred way to self-host wasm files in V2?

**Comment by GordonSmith** [comment]
You don't have to...  Since this was such a pain point for users I switched to a "self extracting" wasm paradigm:
* The wasm file is compressed with zstd
* The compressed file is converted to text using Base91 (similar to Base64) but in theory more compact
* The text is embedded in the bundled JavaScript and "self extracted" on load...
